code/parse/parse_manager.h

```diff
@@ -0,0 +1,60 @@
+#include "globalincs/systemvars.h"
+#include "globalincs/pstypes.h"
+
+enum class Parse_Input_Types : std::uint8_t
+{
+	USAGE_NOTE,
+	STRING,
+	XSTR,
+	INT,
+	INT_LIST,
+	LONG,
+	FLOAT,
+	FLOAT_LIST,
+	DOUBLE,
+	VEC2,
+	VEC3,
+	BOOL,
+	FLAGS,
+	MARKER,
+	MODEL_FILENAME,
+	TEXTURE_FILENAME,
+	SOUND_FILENAME
+};
+
+class parse_item {
+protected:
+	// basic description 
+	SCP_string	_table_string;
+	SCP_string	_description;
+
+	Parse_Input_Types	_type;
+	// rules for this element
+	bool		_required;
+	int			_enforced_count;		// if this is a list that has a specific number of entries, mark it with this variable
+	int			_major_version_number_implemented;
+	int			_minor_version_number_implemented;
+	int			_revision_number_implemented;
+	int			_major_version_number_deprecated;
+	int			_minor_version_number_deprecated;
+	int			_revision_number_deprecated;
```

Comment:
Please initialize these with default values.

Reply:
Default values should now be set in the struct instead of the constructor.